src/topp/DatabaseSuitability.cpp

```diff
@@ -0,0 +1,367 @@
+          if (no_re_rank)
+          {
+            ++count_novo;
+            continue;
+          }
+
+          if (double(top_hit.getMetaValue("MS:1002252")) - double(second_hit.getMetaValue("MS:1002252")) <= cut_off)
+          {
+            ++count_db;
+            ++count_re_ranked;
+          }
+          else ++count_novo;
+        }
+      }
+      else ++count_db; // top hit is db hit
+    }
+
+    // spectra quality
+
+    UInt64 count_novo_seq = 0;
+    set<AASequence> unique_novo;
+
+    for (const auto& pep_id : novo_peps)
+    {
+      if (pep_id.getHits().empty()) continue;
+      ++count_novo_seq;
+      unique_novo.insert(pep_id.getHits()[0].getSequence());
+    }
+
+    //-------------------------------------------------------------
+    // writing output
+    //-------------------------------------------------------------
+
+    double quality = double(count_db) / (count_db + count_novo);
+    double id_rate = double(count_novo_seq) / count_ms2_lvl;
+
+    OPENMS_LOG_INFO << count_db << " top hits that were found in the database." << endl;
+    OPENMS_LOG_INFO << count_novo << " top hits that were only found in the concatenated de novo peptide." << endl;
+    OPENMS_LOG_INFO << count_interest << " times scored a de novo hit just above a database hit. Of those times " << count_re_ranked << " top de novo hits where re-ranked using a decoy cut-off of " << cut_off << endl;
+    OPENMS_LOG_INFO << "Database suitability: " << quality << endl << endl;
+    OPENMS_LOG_INFO << unique_novo.size() << " unique de novo sequences" << endl;
+    OPENMS_LOG_INFO << count_novo_seq << " total de novo sequences" << endl; 
+    OPENMS_LOG_INFO << count_ms2_lvl << " ms2 spectra." << endl;
+    OPENMS_LOG_INFO << "MS2 ID rate: " << id_rate << endl << endl;
+
+    if (!out.empty())
+    {
+      OPENMS_LOG_INFO << "Writting output to: " << out << endl << endl;
+
+      std::ofstream os(out);
+      os.precision(writtenDigits(double()));
+      os << "#top_db_hits\t" << count_db << endl;
+      os << "#top_novo_hits\t" << count_novo << endl;
+      os << "db_suitability\t" << quality << endl;
+      os << "#total_novo_seqs\t" << count_novo_seq << endl;
+      os << "#unique_novo_seqs\t" << unique_novo.size() << endl;
+      os << "#ms2_spectra" << count_ms2_lvl << endl;
+      os << "ms2_id_rate\t" << id_rate << endl;
+      os.close();
+    }
+
+    return EXECUTION_OK;
+
+  }
+
+private:
+
+  double getDecoyDiff_(const PeptideIdentification& pep_id)
+  {
+    double diff = -1;
+
+    // get the score of the first two decoy hits
+    double decoy_1 = -1;
+    double decoy_2 = -1;
+    UInt curr_hit = 1;
+
+    for (const auto& hit : pep_id.getHits())
+    {
+      if (curr_hit > 10) break;
+      ++curr_hit;
+
+      if (!hit.metaValueExists("target_decoy"))
+      {
+        throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No target/decoy information found! Make sure 'PeptideIndexer' is run before hand."));
```

Comment:
FDR needs to run...

Reply:
Not for this meta value though, right? PeptideIndexer should be enough for that.
If you need to ensure that FDR ran, you need to check for a q-value/FDR score (check IDScoreSwitcher findScoreType functionality).